src/Backend/Core/Engine/Meta.php

```diff
@@ -477,6 +477,11 @@ public function save($update = false)
             $custom = null;
         }
 
+        //serialize data for save
+        if (!empty($this->data['data'])) {
+            $this->data['data'] = serialize($this->data['data']);
```

Comment:
you could checkout to master, create a new branch, and then cherry-pick the needed commits in there.